lib/Sema/Constraint.cpp

```diff
@@ -324,8 +327,15 @@ void Constraint::dump(ConstraintSystem *CS) const {
   // Print all type variables as $T0 instead of _ here.
   llvm::SaveAndRestore<bool> X(CS->getASTContext().LangOpts.
                                DebugConstraintSolver, true);
-  
+  // Disabled MSVC warning: only for use within the debugger
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4996)
+#endif
   dump(&CS->getASTContext().SourceMgr);
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif
 }
```

Comment:
Yeah it's not clear to me why the difference exists here. I sometimes hit the deprecation warnings when I add my own calls to dump() in code that I am working on, so I don't know why clang wouldn't emit something here as well, but that's not really your problem.